runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -301,8 +301,125 @@ uint8_t *TR::ARM64CallSnippet::emitSnippetBody()
    return cursor+4;
    }
 
+
+uint8_t *
+TR_Debug::printARM64ArgumentsFlush(TR::FILE *pOutFile, TR::Node *callNode, uint8_t *cursor, int32_t argSize)
+   {
+   uint32_t intArgNum=0, floatArgNum=0, offset;
+   TR::Machine *machine = _cg->machine();
+   TR::Linkage* linkage = _cg->getLinkage(callNode->getSymbol()->castToMethodSymbol()->getLinkageConvention());
+   const TR::ARM64LinkageProperties &linkageProperties = linkage->getProperties();
+   int32_t argStart = callNode->getFirstArgumentIndex();
+   TR::RealRegister *stackPtr = _cg->getStackPointerRegister();
+
+   if (linkageProperties.getRightToLeft())
+      offset = linkage->getOffsetToFirstParm();
+   else
+      offset = argSize+linkage->getOffsetToFirstParm();
+
+   for (int32_t i=argStart; i<callNode->getNumChildren();i++)
+      {
+      TR::Node *child = callNode->getChild(i);
+      switch (child->getDataType())
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+            if (!linkageProperties.getRightToLeft())
+               offset -= TR::Compiler->om.sizeofReferenceAddress();
+            if (intArgNum < linkageProperties.getNumIntArgRegs())
+               {
+               printPrefix(pOutFile, NULL, cursor, 4);
```

Comment:
These printing lines are mostly common across the data type.
How about making a printing function that takes an instruction mnemonic and the register?

Reply:
Created a separate method for printing a single argument store, and changed `printARM64ArgumentsFlush` to use it.